CARDS-1807: Add support for specifying different selectors to use for serializing a subject's forms

sofi2002sofi opened the pull request:
[CARDS-1807](https://phenotips.atlassian.net/browse/CARDS-1807)

To check:
- in proms mode, create Patient Information and Visit Information forms for a patient/visit
- PATIENT.data.dataOption:formSelectors=deep serializes all the answers for the Patient Information form, but does not list the visit subject
- PATIENT.data.deep.dataOption:formSelectors= does not include answers, but includes the visit subject
- PATIENT.data.deep includes both form answers and the visit subject

sofi2002sofi committed:
CARDS-1807: Add support for specifying different selectors to use for serializing a subject's forms

sdumitriu commented:
Setting different form selectors requires escaping the dot with `%5C` like:
`/Subjects/41d2bbf3-54ff-42be-b44f-c85a470a7058.data.dataOption:descendantData=true.dataOption:formSelectors=-dereference%5C.deep.json`.

sdumitriu committed:
CARDS-1807: Add support for specifying different selectors to use for serializing a subject's forms
Use `dataOptions` as the key in the output JSON for mirroring the specified data option parameters
